### PR TITLE
o/registrystate: create modify-registry tasks

### PR DESCRIPTION
**Pull request by MiguelPires**
Adds a helper that creates the appropriate tasks to gather approval, save, commit and notify of a change to a registry. The specific approval gathering mechanism is described in SD133 and I won't describe it here in full since it's somewhat complex.
This PR doesn't include the task/hook handlers that deal with the tasks, it's only about organising the change. Please see the spike https://github.com/canonical/snapd/pull/14283 for how this fits together with the rest of the changes.

### CODE REVIEW

File: overlord/registrystate/registrystate.go [hunk 1]
```diff
@@ -226,6 +228,149 @@ func RegistryTransaction(ctx *hookstate.Context, reg *registry.Registry) (*Trans
...
+	linkTask := func(t *state.Task) {
+		if len(tasks) > 0 {
+			t.WaitFor(tasks[len(tasks)-1])
+		}
+		tasks = append(tasks, t)
+		chg.AddTask(t)
+	}
+
+	// if the transaction errors, clear the tx from the state
+	clearTxOnErrTask := st.NewTask("clear-transaction-on-error", "Clears the ongoing transaction from state (on error)")
```
Comment (pedronis) [inline overlord/registrystate/registrystate.go:260]: s/transaction/registiry-tx/ perhaps, same below

**Review by pedronis — COMMENTED** [review]
did a first pass, this looks reasonable on its own, small comment

File: overlord/registrystate/registrystate.go (hunk 1 above)
Comment (MiguelPires) [reply overlord/registrystate/registrystate.go]: Done

**Review by pedronis — APPROVED** [review]
thank you

File: overlord/registrystate/registrystate.go [hunk 2]
```diff
@@ -226,6 +228,149 @@ func RegistryTransaction(ctx *hookstate.Context, reg *registry.Registry) (*Trans
...
+	// look for plugs that reference the relevant view and create run-hooks for
+	// them, if the snap has those hooks
+	for _, name := range managerNames {
+		plug := managerPlugs[name]
+		manager := plug.Snap
+		if _, ok := manager.Hooks["change-view-"+plug.Name]; !ok {
+			continue
+		}
+
+		ignoreError := false
```
Comment (andrewphelpsj) [inline overlord/registrystate/registrystate.go:272]: nitpick: could be const (and the following ones too)

**Review by andrewphelpsj — APPROVED** [review]
Thanks! lgtm

**Comment by MiguelPires** [comment]
The only failure is `google:ubuntu-20.04-64:tests/main/nvidia-files:535_server` which is unrelated to registries. Also, the code added in this PR is not yet used by the main registries code paths

**Pull request merged**